### ISSUE TITLE
Change Objective function, but very similar results.

### ISSUE THREAD

**Issue by ShiRuiUT** [tensorflow/lucid#136]
Hello, I found that objective function about 
`@wrap_objective
def channel(layer, n_channel, batch=None):` 
only use one activation map value to do optimize. 

So I try to redefine the function from 
`lambda T: tf.reduce_mean(T(layer)[..., n_channel])` 
to
 `    def inner(T):
      a_vecs = tf.unstack(T(layer), axis=-1)
      del a_vecs[n_channel]
      removedTlayer = tf.stack(a_vecs, -1)
      Tlayer = T(layer)[..., n_channel]
      return -tf.reduce_mean(Tlayer) + tf.reduce_mean(tf.abs(tf.reduce_mean(removedTlayer, [0, 1, 2])))
    return inner`
, and change loss to positive in render.py optimization part.
That is mean to optimize image to activate selected activation map and inactive all other activation maps. But the results seems like very similar. 

Would you like share some thoughts about it? 
Or it is just because some properties of tensorflow, e.g. all other activation maps will not be considered to compute gradients to update to the image when objective function is only connected with one activation map. Thank you very much.
(I am newer in tensorflow, in fact, I begin to use it because your great and easy-understanding paper in distill)

**Comment by colah** [comment]
Hi there! I think your proposed objective is equivelant to:

```py
@wrap_objective
def channel_supress_others(layer, n_channel):
  def inner(T):
    channel_act_sum = tf.reduce_sum(T(layer)[..., n_channel])
    layer_act_sum   = tf.reduce_sum(T(layer))
    return -2*channel_act_sum + layer_act_sum
  return inner
```

This kind of objective, where you encourage one neuron to fire and discourage others, is an interesting line of inquiry! You might be interested in this issue on [Feature Visualization Objectives](https://github.com/tensorflow/lucid/issues/116).

In some cases, this might not change things that much. Once the other channels hit zero (their minimum due to ReLU) there won't be an effect. In other cases, it could have a bit effect, if channels are correlated.

Happy exploring!
